task/upload.go

```diff
@@ -244,6 +245,21 @@ func processCatalystCallback(tctx *TaskContext, callback *clients.CatalystCallba
 			})
 		case OutputNameIPFSSourceMP4:
 			assetSpec.Storage.IPFS.CID = output.Manifest
+		case OutputNameAssetMP4:
+			if len(output.Videos) != 1 {
+				return nil, fmt.Errorf("unexpected number of videos in source MP4 output: %d", len(output.Videos))
+			}
+			for _, video := range output.Videos {
+				if video.Type != "mp4" {
+					return nil, fmt.Errorf("unexpected video type in source MP4 output: %s", output.Videos[0].Type)
+				}
+				videoFilePath = video.Location
+				assetSpec.Files = append(assetSpec.Files, api.AssetFile{
+					Type: "static_transcoded_mp4",
+					Path: videoFilePath,
```

Comment:
I actually think we need to prepend this file path now with `mp4`, as we're adding that to the URL sent to catalyst for the output. Bonus if you can make it with a helper or something

Reply:
Are you talking about `videoFilePath` having `mp4/` prepended here?

Reply:
Yeah! Because the destionation we send to catalyst already contains that prefix in relation to the asset folder, so Catlayst should return a path relative to that, meaning we have to add the same prefix here. 

Bonus bonus if you can even use the same helper as for building output_location, but I think the logic is different lol

Reply:
Cool! Done with a separate helper hehe

Reply:
Actually, what you should just use `extractOSUriFilePath` helper function! No need to repeat the `mp4` folder anywhere 😄

Reply:
Mmmm not sure about how `extractOsUriFilePath` works, is this what you intended? https://github.com/livepeer/task-runner/pull/132/commits/82be9ae2ef0b6797234b422f382050abebe250e5

Reply:
Yeah!